### PR TITLE
train acc loss validation acc calculation and tensorboard graphs are added during train

### PR DESCRIPTION
**Pull request by YCAyca**
train loss comes from train dataset, train acc comes from train dataset but in test mode so we dont have data augmentation for this dataset and validation accuracy comes from validation dataset are calculated and logged in tensorboard during train time

### CODE REVIEW

**Comment by MartinHahner** [comment]
@sshaoshuai: do you know someone who can clean and verify this pull request?

**Comment by sshaoshuai** [comment]
Thanks for the PR and discussion!

1. I agree with you that it would be nice to have a choice to get both training loss / validatino loss in the training stage for better comparison. 
2. The main reason that I didn't add it before is that the training process may be blocked after the switch of training/evaluation dataloader (for multi-GPU training), and the evaluation process may also take much longer time for some evaluation tools (like Waymo Open Dataset).  

3. For the PR, I am afraid that I can not merge it at current state since there are many modifications and may destroy the code consistency for multiple datasets. 
Actually it is also very difficult to find someone to clean the PR. 
The easiest way should be that start from  the latest code of OpenPCDet and just add the codes what we need.

4. However, as I mentioned above, this function is not that easy since we need to consider the consistency among different datasets, not just KITTI dataset.  A proper way that I tried before is to add the ``eval_one_epoch`` function here to evaluate the model after each training epoch (might cause blocking during training with multiple-GPU, but it just needs to add very little codes): 
https://github.com/open-mmlab/OpenPCDet/blob/a9c66fee5348dcfb5f73ccca71becc2871ce3a78/tools/train_utils/train_utils.py#L131

Considering the above facts, I do not have plan to merge such a function until it is verified that the training process will not be blocked with multiple GPUs and the codes are clean.

**Comment by sshaoshuai** [comment]
Close it since it seems there are many personal files.
